rest/replicator_test.go

```diff
@@ -3071,3 +3072,335 @@ func TestBlipSyncNonUpgradableConnection(t *testing.T) {
+			remoteRevID: "1-b",
+			conflictResolver: `function(conflict) {
+				var mergedDoc = new Object();
+				mergedDoc.source = "merged";
+				var mergedAttachments = new Object();
+
+				dst = conflict.RemoteDocument._attachments;
+				for (var key in dst) {
+					mergedAttachments[key] = dst[key];
+				}
+				src = conflict.LocalDocument._attachments;
+				for (var key in src) {
+					mergedAttachments[key] = src[key];
+				}
+				mergedDoc._attachments = mergedAttachments;
+				return mergedDoc;
+			}`,
+			expectedLocalBody: map[string]interface{}{
+				db.BodyAttachments: map[string]interface{}{
+					"A": map[string]interface{}{
+						"digest": "sha1-fRV9fAAK4n2xRldcCM4w34k9OmQ=",
+						"length": json.Number("2"),
+						"revpos": json.Number("1"),
+						"stub":   true,
+					},
+					"B": map[string]interface{}{
+						"data":   "Qgo=",
+						"digest": "sha1-MYNq6qsi3ElVWpfttMdTiBQy4B0=",
+						"length": json.Number("2"),
+						"revpos": json.Number("1"),
+					},
+				},
+				"source": "merged",
+			},
+			expectedLocalRevID: createRevID(2, "1-b", db.Body{
+				db.BodyAttachments: map[string]interface{}{
+					"A": map[string]interface{}{
+						"digest": "sha1-fRV9fAAK4n2xRldcCM4w34k9OmQ=",
+						"length": 2,
+						"revpos": 1,
+						"stub":   true,
+					},
+					"B": map[string]interface{}{
+						"data":   "Qgo=",
+						"digest": "sha1-MYNq6qsi3ElVWpfttMdTiBQy4B0=",
+						"length": 2,
+						"revpos": 1,
+					},
+				},
+				"source": "merged",
+			}),
+		},
+		{
+			name: "mergeReadIntlPropsLocalExpiry",
+			localRevisionBody: db.Body{
+				"source":      "local",
+				db.BodyExpiry: "2020-08-24T17:08:29-07:00",
+			},
+			localRevID:         "1-a",
+			remoteRevisionBody: db.Body{"source": "remote"},
+			remoteRevID:        "1-b",
+			conflictResolver: `function(conflict) {
+				var mergedDoc = new Object();
+				mergedDoc.source = "merged";
+				mergedDoc.localDocExp = conflict.LocalDocument._exp;
+				return mergedDoc;
+			}`,
+			expectedLocalBody: db.Body{
+				"localDocExp": "2020-08-24T17:08:29-07:00",
+				"source":      "merged",
+			},
+			expectedLocalRevID: createRevID(2, "1-b", db.Body{
+				"localDocExp": "2020-08-24T17:08:29-07:00",
+				"source":      "merged",
+			}),
+		},
+		{
+			name: "mergeWriteIntlPropsLocalExpiry",
+			localRevisionBody: db.Body{
+				"source":      "local",
+				db.BodyExpiry: "2020-08-24T17:08:29-07:00",
+			},
+			localRevID: "1-a",
+			remoteRevisionBody: db.Body{
+				"source": "remote",
+			},
+			remoteRevID: "1-b",
+			conflictResolver: `function(conflict) {
+				var mergedDoc = new Object();
+				mergedDoc.source = "merged";
+				mergedDoc.localDocExp = "2021-08-24T17:08:29-07:00";
+				return mergedDoc;
+			}`,
+			expectedLocalBody: db.Body{
+				"localDocExp": "2021-08-24T17:08:29-07:00",
+				"source":      "merged",
+			},
+			expectedLocalRevID: createRevID(2, "1-b", db.Body{
+				"localDocExp": "2021-08-24T17:08:29-07:00",
+				"source":      "merged",
+			}),
+		},
```

Comment:
This test doesn't seem to be doing anything different than `mergeReadIntlPropsLocalExpiry`.
Was the intention to set the fixed expiry value in an `_exp` field on the merged doc?

Reply:
Yeah, this was intended to set _exp on the merged doc. Refactored this in latest revision.